src/app/ngrx/effects/cms.effects.ts

```diff
@@ -1,63 +1,97 @@
 import { Injectable } from '@angular/core';
 import { Router, Params, RoutesRecognized } from '@angular/router';
 import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/observable/of';
+import 'rxjs/add/operator/mergeMap';
 import 'rxjs/add/operator/switchMap';
+
 import { Action, Store } from '@ngrx/store';
 import { Actions, Effect } from '@ngrx/effects';
-import { EpisodeModel, EPISODE_PAGE_SIZE, EpisodeMetricsModel } from '../';
-import { selectEpisodeMetrics } from '../reducers';
-import { CmsPodcastEpisodePageAction, CmsEpisodePagePayload,
-  CmsPodcastEpisodePageSuccessAction, CmsPodcastEpisodePageFailureAction, ActionTypes } from '../actions';
-import { getColor, getShade } from '../../shared/util/chart.util';
+
+import { AuthService } from 'ngx-prx-styleguide';
 import { CmsService, HalDoc } from '../../core';
+import { getColor } from '../../shared/util/chart.util';
+
+import { AccountModel, PodcastModel, EpisodeModel, EPISODE_PAGE_SIZE, EpisodeMetricsModel } from '../';
+import { selectEpisodeMetrics } from '../reducers';
+import * as ACTIONS from '../actions';
 
 @Injectable()
 export class CmsEffects {
   episodeMetrics: EpisodeMetricsModel[] = [];
   routeParams: Params;
 
+  @Effect()
+  loadAccount$: Observable<Action> = this.actions$
+    .ofType(ACTIONS.ActionTypes.CMS_ACCOUNT)
+    .switchMap(() => {
+      return this.auth.token.first().mergeMap(token => {
```

Comment:
Had to throw a `first()` in here to make sure this observable actually finishes, and doesn't re-run every time a new token is emitted.

Reply:
👍 that's not one I've had to use before